### PR TITLE
fixes and improvements to FFmpeg autocompletion

### PR DESCRIPTION
**Pull request by jopejoe1**
Things done in this pr:

- fixes format completion, not returning anything
- filter device type for sink and source
- filter codec type
- on codec completion also fill in encoders and copy
- handle subtitle codec
- autocomplete all codecs if codec is not specified

### CODE REVIEW

File: pkg/actions/tools/ffmpeg/codec.go [hunk 1]
```diff
@@ -11,18 +11,48 @@ import (
...
+
+			// copy
+			vals["copy"] = "copy the codec of the input"
+
+			// Parse codecs
+			for _, line := range linesCodecs {
+				if r.MatchString(line) {
+					matches := r.FindStringSubmatch(line)
+					prefix := matches[1]
+					if (codecType == '\u0000' || byte(codecType) == prefix[2]) && 'E' == prefix[1] {
```
Comment (ccoVeille) [inline pkg/actions/tools/ffmpeg/codec.go:33]: All the `\u0000` should use a constant. What it does now in unclear

Comment (rsteube) [reply pkg/actions/tools/ffmpeg/codec.go:33]: It is just whining about the order of the check ([ST1017](https://staticcheck.dev/docs/checks/#ST1017)):

```go
-					if (codecType == '\u0000' || byte(codecType) == prefix[2]) && 'E' == prefix[1] {
+					if (codecType == '\u0000' || byte(codecType) == prefix[2]) && prefix[1] == 'E' {
```

Comment (jopejoe1) [reply pkg/actions/tools/ffmpeg/codec.go]: Reordered the conditions.

Comment (ccoVeille) [reply pkg/actions/tools/ffmpeg/codec.go]: > All the `\u0000` should use a constant. What it does now in unclear

I don't think the point I raised was understood.

I was not talking about a CI complaining about this line.

I was asking to consider adding a constant for \u0000and give to that constant a meaningful name, and a decent comment

Comment (rsteube) [reply pkg/actions/tools/ffmpeg/codec.go]: Ah yes, that is still a bit unclear what it represents.

Comment (rsteube) [reply pkg/actions/tools/ffmpeg/codec.go]: I'll get this in later, just leave it as it is.

**Comment by rsteube** [comment]
Ok, hope I got all the important bits in. Just open another issue/pr if I botched something.

**Pull request merged**